dali/python/nvidia/dali/external_source.py

```diff
@@ -33,9 +33,10 @@ def _check_data_batch(data, batch_size, layout):
         if layout is not None and layout != "" and dim != len(layout):
             raise RuntimeError("The layout '{}' cannot describe {}-dimensional data".format(layout, dim))
 
-class _CycleIter():
-    def __init__(self, iterable):
+class _CycleIter:
+    def __init__(self, iterable, how):
```

Comment:
I feel that `how` is not really descriptive enough. How about `mode`?

Reply:
:+1: - that or just take the `signaling` boolean